sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/TestRegexScannerContext.java

```diff
@@ -0,0 +1,217 @@
+package org.sonar.plugins.secrets.utils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.sonar.api.batch.fs.InputComponent;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.check.Rule;
+import org.sonar.java.regex.RegexCheck;
+import org.sonar.java.regex.RegexScannerContext;
+import org.sonar.plugins.java.api.JavaCheck;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.JavaVersion;
+import org.sonar.plugins.java.api.caching.CacheContext;
+import org.sonar.plugins.java.api.tree.CompilationUnitTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonarsource.analyzer.commons.regex.RegexParseResult;
+import org.sonarsource.analyzer.commons.regex.ast.FlagSet;
+import org.sonarsource.analyzer.commons.regex.ast.RegexSyntaxElement;
+
+public class TestRegexScannerContext implements JavaFileScannerContext, RegexScannerContext {
+
+  private final List<Issue> issues = new ArrayList<>();
+  private String secretRuleKey;
+  private String globalRegexText;
+
+  public void verify() {
+    if (!issues.isEmpty()) {
+      StringBuilder message = new StringBuilder("Found following issues in Regexes:\n");
+      for (Issue issue : issues) {
+        var msg = String.format("%s, the regex: `%s`, violating rule: %s: %s\n",
+          issue.secretRuleKey,
+          issue.regexText,
+          issue.issueRuleKey,
+          issue.message);
+        message.append(msg);
+      }
+      throw new AssertionError(message.toString());
+    }
```

Comment:
The output should include information where we can find the regex. As we don't have the information from which file the rule came, I would add the following information:

- `rule.id`
- Kind / location of pattern (mainDetection, auxiliary / postFilter...)